examples/krypta.rs

```
@@ -884,7 +885,10 @@ fn system_control_mouse_light(
         let ndc_to_world = camera_transform.compute_matrix() * camera.clip_from_view().inverse();
         let mouse_world = ndc_to_world.project_point3(mouse_ndc.extend(-1.0));
 
-        let (mut mouse_transform, mut mouse_color) = query_light.single_mut();
+        let Ok((mut mouse_transform, mut mouse_color)) = query_light.single_mut() else {
+            assert!(false, "Mouse light not found!");
+            return;
+        };
```

You could turn the `Query` into a `Single` and the system should automatically panic with a relevant error if the entity is not present.

---

You could do `query_light.into_inner()` to avoid `ref`, dereferencing, etc. Up to you, I see this pattern is pretty comon in bevy examples when using `Single`.